File: works/2004.kt
import java.util.*
import java.io.*;
import java.io.BufferedWriter;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

fun Cal(n:Long, div:Long): Long
{
    var cnt = 0L
    var _n = n
    while(_n > 0)
    {
        _n /= div
        cnt += _n
    }
    return cnt
}

fun main()
{
    val br = BufferedReader(InputStreamReader(System.`in`))
    var _s = br.readLine().split(" ")
    var n = _s[0].toLong()
    var m = _s[1].toLong()

    var a = Cal(n,2) - (Cal(m,2) + Cal(n-m,2))
    var b = Cal(n, 5) - (Cal(m,5) + Cal(n-m, 5))
    print(Math.min(a,b))
}

File: works/js/3-8/2004.js
let fs = require('fs');
let input = fs.readFileSync(0).toString().trim().split('\n');
// let input = fs.readFileSync('input.txt').toString().split('\n');

function Cal(n, div) {
    let cnt = 0 
    while(n > 0) {
        n = Math.floor(n / div) 
        cnt += n 
    }
    return cnt 
}

let [n, m] = input[0].split(' ').map(Number) 
let a = Cal(n,2) - (Cal(m,2) + Cal(n-m,2));
let b = Cal(n, 5) - (Cal(m,5) + Cal(n-m, 5));
console.log(Math.min(a,b))

  